Plugins/Gnn/include/ActsPlugins/Gnn/Tensor.hpp

```diff
@@ -180,4 +190,38 @@ std::pair<Tensor<std::int64_t>, std::optional<Tensor<float>>> applyEdgeLimit(
     const std::optional<Tensor<float>> &edgeFeatures, std::size_t maxEdges,
     std::optional<cudaStream_t> stream);
 
+template <Acts::Concepts::arithmetic T>
+void Tensor<T>::dumpNpy(const std::string &filename,
+                        std::optional<cudaStream_t> stream) const {
+  std::optional<Tensor<T>> maybeCpuTensor;
+  if (!device().isCpu()) {
+    maybeCpuTensor = this->clone({Device::Cpu(), stream});
+  }
+  const Tensor<T> &toDump =
+      maybeCpuTensor.has_value() ? maybeCpuTensor.value() : *this;
+
+  // Simple NPY header for 2D float32 array
+  std::ofstream ofs(filename, std::ios::binary);
+  if (!ofs.is_open()) {
+    throw std::runtime_error("Could not open file for writing: " + filename);
+  }
```

Comment:
You're not using this stream in this function, as you pass the path to the other function. Consider removing this here or passing the stream below.